File: app/src/main/java/zachwhaley/atwork/AtWorkAlarmReceiver.java
package zachwhaley.atwork;

import android.content.Context;
import android.content.Intent;
import android.support.v4.content.WakefulBroadcastReceiver;

public class AtWorkAlarmReceiver extends WakefulBroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        AtWork.log("AtWorkAlarmReceiver.onReceive");
        Intent alarmService = new Intent(context, AtWorkAlarmService.class);
        startWakefulService(context, alarmService);
    }
}

File: app/src/main/java/zachwhaley/atwork/AtWorkAlarmReceiver.kt
package zachwhaley.atwork

import android.content.Context
import android.content.Intent
import android.support.v4.content.WakefulBroadcastReceiver

class AtWorkAlarmReceiver : WakefulBroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        AtWork.log("AtWorkAlarmReceiver.onReceive")

        val alarmService = Intent(context, AtWorkAlarmService::class.java)
        alarmService.action = intent.action
        WakefulBroadcastReceiver.startWakefulService(context, alarmService)
    }
}